docs/rules/object-property-newline.md

```diff
@@ -86,28 +90,43 @@ const newObject = {
 
 will be prohibited, because two properties, but not all properties, appear on the same line.
 
-### Notations
-
-This rule applies equally to all property specifications, regardless of notation, including:
+This option also has an older, and now deprecated, name, `allowMultiplePropertiesPerLine`.
 
-- `a: 1` (ES5)
-- `a` (ES2015 shorthand property)
-- ``[`prop${a}`]`` (computed property name)
+#### `treatComputedPropertiesLikeJSCS`
 
-Thus, the rule (without the object option) prohibits both of these:
+If you set `treatComputedPropertiesLikeJSCS` to `true`, an object literal such as the one below will be permitted:
 
 ```js
 const newObject = {
     a: 1, [
         process.argv[4]
     ]: '01'
 };
-const newObject = {
-    a: 1, [process.argv[4]]: '01'
-};
 ```
 
-(This behavior differs from that of the JSCS rule cited below, which does not treat the leading `[` of a computed property name as part of that property specification. The JSCS rule prohibits the second of these formats but permits the first.)
+Otherwise, this rule will prohibit it, because ESLint treats the opening bracket of a computed property name as part of the property specification. The JSCS rule `requireObjectKeysOnNewLine` does not, so this object option makes ESLint compatible with JSCS in this respect.
+
+#### `noCommaFirst`
```

Comment:
is this option new added?
the rulename `object-newline-`, so it might not be the rule's responsibility.(and we already have a rule `comma-style` for this).

Reply:
Yes, this option has been implemented and tested. Its implementation and its tests are included in this pull request.

Thank you, @Aladdin-ADD, for calling attention to the `comma-style` rule. Here are my thoughts about this question:

1. You are correct that the `comma-style` rule deals with the same problem as the `noCommaFirst` object option in this PR. In principle, it would be possible to use the `comma-style` rule in combination with this `object-property-newline` rule to emulate the JSCS `requireObjectKeysOnNewLine` rule.

2. However, there seem to be good reasons not to do that:

2a. The main purpose of this issue (refs #6251) is to make the ESLint `object-property-newline` rule compatible with 2 rules of JSCS.

2b. It would be difficult to use `comma-style` for this purpose. The user would be required to invoke both rules, and, in invoking the `comma-style` (`last`) rule, would need to invoke its `exceptions` object with 9 enumerated exceptions, to exclude all applications of the rule except `ObjectExpression`.

2c. Even that would not produce full compatibility. The combination of object options `noCommaFirst` and `treatComputedPropertiesLikeJSCS` makes the pattern `foo: 'bar'\n, [\nfoo + 'key']: 'baz'` valid. But the combination of `treatComputedPropertiesLikeJSCS` with the `comma-style` (`last`) rule would make it invalid.

2d. The combination of these two rules would produce stylistically inconsistent error messages. This rule reports line and column numbers, but the `comma-style` rule does not.

I look forward to your thoughs about these concerns.